arkouda/pdarrayclass.py

```diff
@@ -1772,6 +1809,214 @@ def argmaxk(pda : pdarray, k : int_scalars) -> pdarray:
+    array([0, 1, 1, 0, 1, 0, 0, 1, 1, 0])
+    """
+    if pda.dtype != akint64:
+        raise TypeError("BitOps only supported on int64 arrays")
+    repMsg = generic_msg(cmd="efunc", args="{} {}".format("parity", pda.name))
+    return create_pdarray(repMsg)
+
+def clz(pda: pdarray) -> pdarray:
+    """
+    Count leading zeros for each integer in an array.
+
+    Parameters
+    ----------
+    pda : pdarray, int64
+        Input array (must be integral).
+
+    Returns
+    -------
+    lz : pdarray
+        The number of leading zeros of each element.
+
+    Raises
+    ------
+    TypeError
+        If input array is not int64
+    
+    Examples
+    --------
+    >>> A = ak.arange(10)
+    >>> ak.clz(A)
+    array([64, 63, 62, 62, 61, 61, 61, 61, 60, 60])
+    """
+    if pda.dtype != akint64:
+        raise TypeError("BitOps only supported on int64 arrays")
+    repMsg = generic_msg(cmd="efunc", args="{} {}".format("clz", pda.name))
+    return create_pdarray(repMsg)
+
+def ctz(pda: pdarray) -> pdarray:
+    """
+    Count trailing zeros for each integer in an array.
+
+    Parameters
+    ----------
+    pda : pdarray, int64
+        Input array (must be integral).
+
+    Returns
+    -------
+    lz : pdarray
+        The number of trailing zeros of each element.
+
+    Notes
+    -----
+    ctz(0) is defined to be zero.
+
+    Raises
+    ------
+    TypeError
+        If input array is not int64
+    
+    Examples
+    --------
+    >>> A = ak.arange(10)
+    >>> ak.ctz(A)
+    array([0, 0, 1, 0, 2, 0, 1, 0, 3, 0])
+    """
+    if pda.dtype != akint64:
+        raise TypeError("BitOps only supported on int64 arrays")
+    repMsg = generic_msg(cmd="efunc", args="{} {}".format("ctz", pda.name))
+    return create_pdarray(repMsg)
+
+def rotl(x, rot) -> pdarray:
+    """
+    Rotate bits of <x> to the left by <rot>.
+
+    Parameters
+    ----------
+    x : pdarray(int64) or integer
+        Value(s) to rotate left.
+    rot : pdarray(int64) or integer
+        Amount(s) to rotate by.
+
+    Returns
+    -------
+    rotated : pdarray(int64)
+        The rotated elements of x.
+
+    Raises
+    ------
+    TypeError
+        If input array is not int64
+    
+    Examples
+    --------
+    >>> A = ak.arange(10)
+    >>> ak.rotl(A, A)
+    array([0, 2, 8, 24, 64, 160, 384, 896, 2048, 4608])
+    """
+    if isinstance(x, pdarray):
+        if x.dtype != akint64:
+            raise TypeError("Rotations only supported on integers")
+        if (isinstance(rot, pdarray) and rot.dtype == akint64) or isSupportedInt(rot):
+            return x._binop(rot, "<<<")
+        else:
+            raise TypeError("Rotations only supported on integers")
+    elif isSupportedInt(x):
+        if isinstance(rot, pdarray) and rot.dtype == akint64:
+            return rot._r_binop(x, "<<<")
+        else:
+            raise TypeError("Rotations only supported on integers")
+    else:
+        raise TypeError("Rotations only supported on integers")
```

Comment:
This is just stylistic but I think restructuring the if/else statements here makes it more readable. Something similar can be done to `rotr`. No need to change it
```python
    if isinstance(x, pdarray) and x.dtype == akint64:
        if (isinstance(rot, pdarray) and rot.dtype == akint64) or isSupportedInt(rot):
            return x._binop(rot, "<<<")
        else:
            raise TypeError("Rotations only supported on integers")
    elif isSupportedInt(x) and isinstance(rot, pdarray) and rot.dtype == akint64:
        return rot._r_binop(x, "<<<")
    else:
        raise TypeError("Rotations only supported on integers")
```

Reply:
That's a good suggestion -- implemented!